lib/ansible/modules/cloud/vmware/vmware_guest_disk_facts.py

```diff
@@ -0,0 +1,194 @@
+     required: True
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: Gather disk facts from virtual machine using UUID
+  vmware_guest_disk_facts:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    datacenter: ha-datacenter
+    validate_certs: no
+    uuid: 421e4592-c069-924d-ce20-7e7533fab926
+  delegate_to: localhost
+  register: disk_facts
+
+- name: Gather disk facts from virtual machine using name
+  vmware_guest_disk_facts:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    datacenter: ha-datacenter
+    validate_certs: no
+    name: VM_225
+  delegate_to: localhost
+  register: disk_facts
+'''
+
+RETURN = """
+guest_disk_facts:
+    description: metadata about the virtual machine's disks
+    returned: always
+    type: dict
+    sample: {
+        "0": {
+            "backing_datastore": "datastore2",
+            "backing_disk_mode": "persistent",
+            "backing_eagerlyscrub": false,
+            "backing_filename": "[datastore2] VM_225/VM_225.vmdk",
+            "backing_thinprovisioned": false,
+            "backing_writethrough": false,
+            "capacity_in_bytes": 10485760,
+            "capacity_in_kb": 10240,
+            "controller_key": 1000,
+            "key": 2000,
+            "label": "Hard disk 1",
+            "summary": "10,240 KB",
+            "unit_number": 0
+        },
+    }
+"""
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec
+
+
+class PyVmomiHelper(PyVmomi):
+    def __init__(self, module):
+        super(PyVmomiHelper, self).__init__(module)
+
+    def gather_disk_facts(self, vm_obj):
+        """
+        Function to gather facts about VM's disks
+        Args:
+            vm_obj: Managed object of virtual machine
+
+        Returns: A list of dict containing disks information
+
+        """
+        disks_facts = dict()
+        if vm_obj is None:
+            return disks_facts
+
+        disk_index = 0
+        for disk in vm_obj.config.hardware.device:
+            if isinstance(disk, vim.vm.device.VirtualDisk):
+                disks_facts[disk_index] = dict(
+                    key=disk.key,
+                    label=disk.deviceInfo.label,
+                    summary=disk.deviceInfo.summary,
+                    backing_filename=disk.backing.fileName,
+                    backing_datastore=disk.backing.datastore.name,
+                    backing_disk_mode=disk.backing.diskMode,
+                    backing_writethrough=disk.backing.writeThrough,
+                    backing_thinprovisioned=disk.backing.thinProvisioned,
+                    backing_eagerlyscrub=disk.backing.eagerlyScrub if disk.backing.eagerlyScrub else False,
```

Comment:
`backing_eagerlyscrub=bool(disk.backing.eagerlyScrub),`
?

Reply:
👍 
This will happen when you overthink.